### ISSUE TITLE
templates have some issues

### ISSUE THREAD

**Issue by despiegk** [vlang/v#8676]
# crash

```
out := $tmpl('index.html')
```

does not work but following works

```
fn index_template(wikis []string) string{
	return $tmpl('index.html')
}
```

# how to specify location of template

$tmpl('index.html') only works if the template is in path where v run ... is called
not local to where the file was where this statement is (module path)

there should be more flexibility to define where the template is, @FILE also does not work we need @DIR or so

**Comment by changrui** [comment]
```
module main
fn main() {
	embedded_file := $embed_file('v.png')
	mut fw := os.create('exported.png') or { panic(err) }
	fw.write_bytes(embedded_file.data(), embedded_file.len)
	fw.close()
}
```

**Comment by changrui** [comment]
```
mut out := $embed_file('a.v')
println(out.len)
println(out.data().vstring())
```

**Comment by Dialga** [comment]
Fixed by #11541.
